src/server/status_server/mod.rs

```diff
@@ -750,6 +750,9 @@ where
                             (Method::GET, "/debug/pprof/profile") => {
                                 Self::dump_rsperf_to_resp(req).await
                             }
+                            (Method::GET, "/debug/panic") => {
```

Comment:
Let's don't add such an API. For debugging we can add this code out of tree.

Reply:
What do you mean by add out of tree?

Another approach would be to set an environment variable, something like `TIKV_DEV_ENV=1` to enable development/testing non-production features.

Reply:
or can we have this available only in dev build? as you said, it provide no usage out of dev environment and prone to attack.

Reply:
like, whoever want to do the testing can add the code themselves locally. I would prefer not to check it in.

Reply:
how should I make it available only in the dev build? Another option is to use a feature flag.

Reply:
If you wish to cause panic, you can run tikv-server-fail-point and use curl to trigger panic by putting instructions to `/fail/failpointname`, any existing frequent access failpoint name should be sufficient.

Reply:
I can try that. Thinking about this more, I think a panic API in a deployed TiKV can be useful for our chaos testing and for customers to harden their deployment before a production launch (they should check that they get alerted when TiKV crashes). I think it could be guarded by a configuration item "debug-panic-api-enable". What do you think?

Reply:
I don't think it's a good idea. It doesn't feel right to expose an API to allow triggering panic in a release TiKV. For chaos testing, failpoints are sufficient.

Reply:
Okay, failpoints works very well for development. I changed the API to have a fail point.

As a best practice I always test the response of a production service (at least on a staging environment) for exception/panic to see that this is monitored. It is a shame there isn't a simple way for TiDB users to do this and they have to re-compile TiKV.

Reply:
I think what @BusyJay mean is remove this new API and use the failpoint API, which is gate behind the "failpoints" feature. https://github.com/tikv/tikv/blob/master/src/server/status_server/mod.rs#L715

Reply:
I changed the new API to use the failpoints features. When compiled with the failpoints feature, it is easy to trigger a panic now. I added this snippet to the PR description as well.

```
curl localhost:20180/fail/debug_fail_point -X PUT -d 'panic'
curl localhost:20180/debug/fail_point
```

Reply:
This new API is redundant since we have the failpoint API. Please remove it.

Reply:
I discussed this with @yiwu-arbug. There is no redundancy here: this API is a new feature that does not exist in the failpoint API.